### PR TITLE
Implement Rotate, mission_wp and detour_wp - v2

### PR DESCRIPTION
**Pull request by guiccbr**
Also implemented brake mode.

### CODE REVIEW

File: tests/tests.cc [hunk 1]
```diff
@@ -118,40 +128,191 @@ void connection_test::show_mav_state()
...
+    this->mav->send_mission_waypoint(target_global);
+
+    // Wait for 5 seconds
+    std::this_thread::sleep_for(std::chrono::duration<int, std::milli>(5000));
+
+    // Send Rotation command
+    this->mav->brake(true);
+
+    // Wait until the vehicle gets to the mission target
+    while (true) {
```
Comment (dakerfp) [inline tests/tests.cc:308]: Why don't you put a condition variable and exit the loop in that case?

File: tests/tests.cc [hunk 2]
```diff
@@ -118,40 +128,191 @@ void connection_test::show_mav_state()
...
+    if (sock == -1) {
+        perror("error opening socket");
+        exit(EXIT_FAILURE);
+    }
+
+    local_addr.sin_family = AF_INET;
+    local_addr.sin_addr.s_addr = htonl(INADDR_LOOPBACK);
+    local_addr.sin_port = htons(defaults::mavproxy_port);
+
+    if (bind(sock, (struct sockaddr *)&local_addr, sizeof(struct sockaddr)) ==
```
Comment (dakerfp) [inline tests/tests.cc:232]: br in a more proper place

File: mavlink_vehicles/mavlink_vehicles.hh [hunk 3]
```diff
@@ -79,17 +110,45 @@ class mav_vehicle
...
+    void rotate(double angle_deg);
+    bool is_rotation_active() const;
+
+    // Command the vehicle to immediately take a detour through the given
+    // waypoint before moving to the next mission waypoint. If the vehicle is
+    // currently currently rotating because of a previous rotate() command, the
+    // rotation will be immediately aborted.
+    void send_detour_waypoint(double lat, double lon, double alt);
+    void send_detour_waypoint(global_pos_int global);
+    void send_detour_waypoint(global_pos_int global, bool autocontinue);
```
Comment (dakerfp) [inline mavlink_vehicles/mavlink_vehicles.hh]: Avoid duplicate send_detour_waypoint by setting default autocontinue to true

File: mavlink_vehicles/mavlink_vehicles.hh [hunk 4]
```diff
@@ -43,29 +44,59 @@ struct attitude : state_variable {
...
 
 struct global_pos_int : state_variable {
-    int32_t lat;
-    int32_t lon;
-    int32_t alt;
+    int32_t lat; // Degrees * 1e7
+    int32_t lon; // Degrees * 1e7
+    int32_t alt; // Millimeters (AMSL)
+    global_pos_int() {}
+    global_pos_int(int32_t _lat, int32_t _lon, int32_t _alt)
```
Comment (dakerfp) [inline mavlink_vehicles/mavlink_vehicles.hh:51]: We couldn't simply use the struct  {} initializer?

File: tests/tests.cc (hunk 1 above)
Comment (guiccbr) [reply tests/tests.cc:308]: Fixed in 1cd47ad223f482e8c667926dbe721528677e7cd6

File: tests/tests.cc (hunk 2 above)
Comment (guiccbr) [reply tests/tests.cc:232]: Fixed in 4070f0725493ac20cfabfb81682180c194a0d587

File: mavlink_vehicles/mavlink_vehicles.hh (hunk 3 above)
Comment (guiccbr) [reply mavlink_vehicles/mavlink_vehicles.hh]: Tks, fixed in 88de5632e85aec4d8585548408998cc078ccfb3f

File: mavlink_vehicles/mavlink_vehicles.hh (hunk 4 above)
Comment (guiccbr) [reply mavlink_vehicles/mavlink_vehicles.hh:51]: Unfortunately this is not possible because global_pos_int is inheriting from state_variable.

**Comment by dakerfp** [comment]
LGTM

**Pull request merged**